### DISCUSSION TITLE
Export equations as image

### DISCUSSION THREAD

**Discussion by denislenz** [KaTeX/KaTeX#3582] [Q&A]
Is there any way to export the html of an equation or the TEX as an image?

My problem is: I have the equations in a serverside environment where I can't use all the css features and I need to get a correct view of the equations to export as DOCX

**Comment by mbourne** [comment]
Some non-KaTeX solutions for you to consider:

1. CodeCogs - https://editor.codecogs.com/ You can save the images (or link to them), up to 1000 per day on the free account. Depending on the number of equations you are dealing with, the best would be to save the generated equations to your server. This is the easiest option, but negatives include having to align each inline equation with the baseline of the rest of your text.
2. Tex2png - https://github.com/Gregwar/Tex2png No restrictions with this one, but the server setup can be onerous. This also has the same issue where you need to align any inline math.

**Comment by edemaine** [comment]
You could take a look at how KaTeX's screenshotter (under `dockers/screenshotter`) does it, using Selenium.  (You'd probably want to use higher resolution than our screenshotter does though.)
